### ISSUE TITLE
Zip and Tar operations should support symbolic links

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle#10676]
Pretty much every native team produces or consumes some kind of tar ball or zip of stuff, such as an application installation distribution or a library SDK or source files or a tool chain or whatever, and in native land these archives often contain symbolic links. The lack of support for symbolic links in Gradle's zip and tar operations (tasks, file trees, etc) mean that these teams need to call out to other tools to produce/consume these archives. These tools are platform specific, need to be installed and are generally a pain to deal with.

Gradle should understand symbolic links in tar and zip files.

It should also provide conventions for producing these kinds of packages from the binaries produced for the project, but that's really another issue.

**Comment by ericsun2** [comment]
Can we please reopen this issue if the problem remains unresolved?
The Unix symbolic link support should be a no brainer for this operation.

**Comment by aalmiray** [comment]
Just came across this issue today. Any news?

**Comment by imavroukakis** [comment]
Hi folks, is there any intent to work on this? Ticket has been open for year, a definitive answer would be appreciated 😄

**Comment by jbartok** [comment]
Came up in:
* https://github.com/gradle/gradle/pull/23470

**Comment by jbartok** [comment]
Thank you for your interest in Gradle!

This feature request is in the backlog of the relevant team but this area of Gradle is currently not a focus one so it might take a while before it gets implemented.

**Comment by martinda** [comment]
Broken symbolic links also need to be supported, in the sense that they should be copied without trying to resolve or follow them. When they unzip, they should be restored as well, even if they are still broken. They will work if the missing files pointed to by the links are restored by some external method prior or after the unzip.

**Comment by sschuberth** [comment]
> Came up in:
> 
>     * [JdkCacheDirectory: Handle duplicates when unpacking JDKs #23470](https://github.com/gradle/gradle/pull/23470)

I belive the underlying issue actually is https://github.com/gradle/gradle/issues/3982 from year 2004...

**Comment by sschuberth** [comment]
FYI @ov7a, cross-linking to @big-guy's comment in an old proposed PR how [preserving symlinks](https://github.com/gradle/gradle/pull/12795#issuecomment-621631469) could be implemented.

**Comment by sschuberth** [comment]
> The zip program supports an option to store symlinks. Could it be that simple?

No, that doesn't help, as Gradle still doesn't know how to restore these links when unpacking.

**Comment by cbfiddle** [comment]
I'd say it helps some people, who want a proper zip file and don't need Gradle to unpack it.

**Comment by MLNW** [comment]
It's pretty crazy that this functionality hasn't been added in six years. Is it at least clear where the changes would have to be made to allow this?

**Comment by sschuberth** [comment]
@MLNW https://github.com/gradle/gradle/pull/25264 lays the foundation for it, looks like it could land for Gradle 8.7.
